packages/meditrak-server/src/dataAccessors/editSurvey.js

```diff
@@ -88,16 +89,25 @@ class SurveyEditor {
     const updatedFields = this.updatedFieldsByResource[resourceType];
     const isDataSource = model.databaseType === this.models.dataSource.databaseType;
 
-    Object.entries(updatedFields).forEach(([fieldName, fieldValue]) => {
-      if (isDataSource && (fieldName === 'config' || fieldValue === undefined)) {
-        // Handle config last since its value depends on other fields
-        // For the same reason, do not set model fields to `undefined`
-        // so that the actual record values can be used for config calculation
-        return;
-      }
-      // eslint-disable-next-line no-param-reassign
-      model[fieldName] = fieldValue;
-    });
+    Object.entries(updatedFields)
+      .filter(([, fieldValue]) => fieldValue !== undefined)
```

Comment:
In the past we'd let `undefined` fields in the model, since they won't actually be updated in the db. We'd still skip those for the data source model, because setting its fields to `undefined` caused issues with using the model state to draw conclusions (see lines 92-96 in existing code)

This causes an issue for the `survey` model too now, so I decided it's high time to just skip those fields!